independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartProcessor.java

```diff
@@ -0,0 +1,159 @@
+                // TODO print a warning to inform that this codestart miss this language
+                return null; // We ignore that codestart as it doesn't implement the specified language
+            }
+            Stream.of(baseDir, languageDir)
+                    .filter(Files::isDirectory)
+                    .forEach(dirPath -> processCodestartDir(dirPath,
+                            CodestartData.buildCodestartData(codestart, languageName, data)));
+            return null;
+        });
+    }
+
+    void processCodestartDir(final Path sourceDirectory, final Map<String, Object> finalData) {
+        try {
+            final Collection<Path> sources = Files.walk(sourceDirectory)
+                    .filter(path -> !path.equals(sourceDirectory))
+                    .collect(Collectors.toList());
+            for (Path sourcePath : sources) {
+                final Path relativeSourcePath = sourceDirectory.relativize(sourcePath);
+                if (!Files.isDirectory(sourcePath)) {
+                    final String sourceFileName = sourcePath.getFileName().toString();
+
+                    // Read files to process
+                    final Optional<CodestartFileReader> possibleReader = CodestartFileReader.ALL.stream()
+                            .filter(r -> r.matches(sourceFileName))
+                            .findFirst();
+                    final CodestartFileReader reader = possibleReader.orElse(CodestartFileReader.DEFAULT);
+
+                    final String targetFileName = reader.cleanFileName(sourceFileName);
+                    final Path relativeTargetPath = relativeSourcePath.getNameCount() > 1
+                            ? relativeSourcePath.getParent().resolve(targetFileName)
+                            : Paths.get(targetFileName);
+
+                    final boolean hasConflictStrategyHandler = getStrategy(relativeTargetPath.toString()).isPresent();
+
+                    if (!possibleReader.isPresent() && !hasConflictStrategyHandler) {
+                        // Copy static files
+                        // TODO support static files from codestart.yml
+                        final Path targetPath = targetDirectory.resolve(relativeTargetPath.toString());
+                        processStaticFile(sourcePath, targetPath);
+                    }
+
+                    final Optional<String> content = reader.read(sourcePath,
+                            languageName, finalData);
+                    if (content.isPresent()) {
+                        final String key = relativeTargetPath.toString();
+                        this.files.putIfAbsent(key, new ArrayList<>());
+                        this.files.get(key).add(new CodestartFile(relativeSourcePath.toString(), content.get()));
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
add/move this try/catch inside the for loop so you can give some context to this exception (like which path is failing on)

Reply:
good catch.. very smart!